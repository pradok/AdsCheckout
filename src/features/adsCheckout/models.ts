import {customerModels} from '../customer';

export type Product = {
  id?: number | string;
  price: number;
  quantity: number;
  priceReduced: number;
  title: string;
};

export type Discount = {
  premium?: customerModels.DiscountPremium;
  classic?: customerModels.DiscountClassic;
  standout?: customerModels.DiscountStandOut;
  rulesDescription: string[];
};
