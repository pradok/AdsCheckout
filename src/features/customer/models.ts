export type DiscountPremium = {
  perAd: {
    minQty: number;
    price: number;
  };
};

export type DiscountStandOut = {
  price: number;
};

export type DiscountClassic = {
  trigger: number;
};
