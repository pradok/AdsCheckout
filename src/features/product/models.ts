export type Product = {
  id: string;
  title: string;
  price: number;
};

export interface ProductCollectionState {
  collection: Product[];
  isLoading: boolean;
  isFetched: boolean;
}
export interface ProductAction {
  type: string;
  payload?: Product[];
}
