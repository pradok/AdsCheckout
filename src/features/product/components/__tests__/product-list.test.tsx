import * as React from 'react';
import {shallow} from 'enzyme';

import ProductList from '../product-list';
import {Product} from '../../models';

const products: Product[] = [
  {
    id: 'classic',
    title: 'Classic Ad',
    price: 269.99,
  },
  {
    id: 'standout',
    title: 'Standout Ad',
    price: 322.99,
  },
  {
    id: 'premium',
    title: 'Premium Ad',
    price: 394.99,
  },
];

const addToCartMock = jest.fn();

it('<ProductList> renders correctly', () => {
  const component = shallow(<ProductList products={products} addToCart={addToCartMock} />);
  expect(component).toMatchSnapshot();
});
