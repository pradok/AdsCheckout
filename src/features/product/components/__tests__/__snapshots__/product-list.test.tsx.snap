// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductList> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductList
    addToCart={[MockFunction]}
    products={
      Array [
        Object {
          "id": "classic",
          "price": 269.99,
          "title": "Classic Ad",
        },
        Object {
          "id": "standout",
          "price": 322.99,
          "title": "Standout Ad",
        },
        Object {
          "id": "premium",
          "price": 394.99,
          "title": "Premium Ad",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Ad packages
        </h2>,
        Array [
          <ProductItem
            onClickAddToCart={[Function]}
            product={
              Object {
                "id": "classic",
                "price": 269.99,
                "title": "Classic Ad",
              }
            }
          />,
          <ProductItem
            onClickAddToCart={[Function]}
            product={
              Object {
                "id": "standout",
                "price": 322.99,
                "title": "Standout Ad",
              }
            }
          />,
          <ProductItem
            onClickAddToCart={[Function]}
            product={
              Object {
                "id": "premium",
                "price": 394.99,
                "title": "Premium Ad",
              }
            }
          />,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Ad packages",
        },
        "ref": null,
        "rendered": "Ad packages",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": "classic",
        "nodeType": "function",
        "props": Object {
          "onClickAddToCart": [Function],
          "product": Object {
            "id": "classic",
            "price": 269.99,
            "title": "Classic Ad",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "standout",
        "nodeType": "function",
        "props": Object {
          "onClickAddToCart": [Function],
          "product": Object {
            "id": "standout",
            "price": 322.99,
            "title": "Standout Ad",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "premium",
        "nodeType": "function",
        "props": Object {
          "onClickAddToCart": [Function],
          "product": Object {
            "id": "premium",
            "price": 394.99,
            "title": "Premium Ad",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Ad packages
          </h2>,
          Array [
            <ProductItem
              onClickAddToCart={[Function]}
              product={
                Object {
                  "id": "classic",
                  "price": 269.99,
                  "title": "Classic Ad",
                }
              }
            />,
            <ProductItem
              onClickAddToCart={[Function]}
              product={
                Object {
                  "id": "standout",
                  "price": 322.99,
                  "title": "Standout Ad",
                }
              }
            />,
            <ProductItem
              onClickAddToCart={[Function]}
              product={
                Object {
                  "id": "premium",
                  "price": 394.99,
                  "title": "Premium Ad",
                }
              }
            />,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Ad packages",
          },
          "ref": null,
          "rendered": "Ad packages",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": "classic",
          "nodeType": "function",
          "props": Object {
            "onClickAddToCart": [Function],
            "product": Object {
              "id": "classic",
              "price": 269.99,
              "title": "Classic Ad",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "standout",
          "nodeType": "function",
          "props": Object {
            "onClickAddToCart": [Function],
            "product": Object {
              "id": "standout",
              "price": 322.99,
              "title": "Standout Ad",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "premium",
          "nodeType": "function",
          "props": Object {
            "onClickAddToCart": [Function],
            "product": Object {
              "id": "premium",
              "price": 394.99,
              "title": "Premium Ad",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
